crypto/ec/ecp_sm2p256.c

```diff
@@ -40,28 +40,28 @@ typedef struct {
 
 #if !defined(OPENSSL_NO_SM2_PRECOMP)
 /* Coordinates of G, for which we have precomputed tables */
-static const BN_ULONG def_xG[P256_LIMBS] ALIGN32 = {
+ALIGN32 static const BN_ULONG def_xG[P256_LIMBS] = {
```

Comment:
Is this likely to break GCC/CLANG?

Reply:
All I can say is that this patch didn't cause build problems in vcpk CI for x64-linux, x64-osx and android (x64, arm, arm64). This covers some versions of gcc and clang. I don't how different versions of gcc or clang handle the attribute.

Reply:
Related existing pattern:
https://github.com/openssl/openssl/blob/e4542332fa36eab6d6bbf33815bde433ade3b547/crypto/bn/rsaz_exp.c#L39-L47

Reply:
I can't see how this would be a problem, GCC has always accepted attributes here.

Reply:
The fix is for VC-WIN64-CLANGASM-ARM. The preprocessor is `cl`, not gcc or clang.

Reply:
Sorry, this is the C file, so the answer is: The *compiler* is `cl`, not gcc or clang.

Reply:
I think this is fine